app/controllers/frontend_log_controller.rb

```diff
@@ -50,6 +50,6 @@ def valid_event?
   end
 
   def valid_payload?
-    !log_params[:payload].nil?
+    params[:payload].nil? || !log_params[:payload].nil?
```

Comment:
I think we should keep the server strict about these

Reply:
The payload is optional for both the Ruby and JavaScript versions of this API. Any reason it shouldn't also be optional for the JSON API?

Reply:
I guess that's right, but all events have extra attributes. I take my comment back

Reply:
I'm not super happy with how the validation turned out here, since how Rails `permit` handling with a hash makes it so all invalid parameters (including a `nil` value) would be normalized to an empty hash, so it requires this juggling of both `params` and `#permit`'d `log_params` to allow omission while still rejecting anything other than a hash.